src/qmapshack/gis/fit/decoder/IFitDecoderState.cpp

```diff
@@ -117,23 +117,22 @@ void IFitDecoderState::incFileBytesRead()
 
 void IFitDecoderState::addDevFieldProfile(const CFitFieldProfile &fieldProfile)
 {
-    // for documentation: a development field definition is linked to an developer data ID. Only the tuple developer data index
-    // and field definition number must be unique. So far no fit file with more than one developer data ID has been created.
-    if(devFieldProfile(fieldProfile.getFieldDefNum())->getFieldDefNum() ==  fieldProfile.getFieldDefNum())
-    {
+    // for documentation: a development field definition is linked to an developer data ID. The tuple developer data index
+    // and field definition number must be unique.
+    if(devFieldProfile(fieldProfile.getDevProfileId())->getDevProfileId() ==  fieldProfile.getDevProfileId())    {
         throw tr("FIT decoding error: a development field with the field_definition_number %1 already exists.")
               .arg(fieldProfile.getFieldDefNum());
     }
     data.devFieldProfiles.append(fieldProfile);
 }
 
-CFitFieldProfile* IFitDecoderState::devFieldProfile(quint32 fieldNr)
+CFitFieldProfile* IFitDecoderState::devFieldProfile(QPair<quint8,quint8> devProfileId)
 {
-    for (int i = 0; i < data.devFieldProfiles.size(); i++)
+    for( auto &devFieldPro : data.devFieldProfiles)
```

Comment:
We try to avoid `auto` as it makes the code hard to read. Not all IDEs/editors resolve auto to the correct type.

Reply:
OK, I removed it at  both places where I used it.